protocol/messenger_handler.go

```diff
@@ -3676,6 +3691,40 @@ func (m *Messenger) handleSyncKeypair(message *protobuf.SyncKeypair, fromLocalPa
 	if err != nil {
 		return nil, err
 	}
+
+	if m.config.accountsFeed != nil {
+		addedAddresses := []gethcommon.Address{}
+		removedAddresses := []gethcommon.Address{}
+		if dbKeypair.Removed {
+			for _, acc := range dbKeypair.Accounts {
+				removedAddresses = append(removedAddresses, gethcommon.Address(acc.Address))
+			}
+			m.config.accountsFeed.Send(accountsevent.Event{
+				Type:     accountsevent.EventTypeRemoved,
+				Accounts: removedAddresses,
+			})
+		} else {
+			for _, acc := range dbKeypair.Accounts {
+				if acc.Chat {
+					continue
+				}
+				if acc.Removed {
+					removedAddresses = append(removedAddresses, gethcommon.Address(acc.Address))
+				} else {
+					addedAddresses = append(addedAddresses, gethcommon.Address(acc.Address))
+				}
+			}
+			m.config.accountsFeed.Send(accountsevent.Event{
```

Comment:
move these two to the bottom and only send the event if len(addedAddresses)>0 and len(removedAddresses)>0, accordingly. That way you also don't duplicate code for `EventTypeRemoved`